cloud/pkg/edgecontroller/controller/upstream.go

```diff
@@ -667,6 +676,92 @@ func (uc *UpstreamController) processServiceAccountToken() {
 	}
 }
 
+func (uc *UpstreamController) processLease() {
+	for {
+		select {
+		case <-beehiveContext.Done():
+			klog.Warning("stop process lease")
+			return
+		case msg := <-uc.leaseChan:
+			content, err := msg.GetContentData()
+			if err != nil {
+				klog.Errorf("failed to get content data from lease message: %s, %v", msg.String(), err)
+				continue
+			}
+			lease := &coordinationv1.Lease{}
+			if err = json.Unmarshal(content, lease); err != nil {
+				klog.Errorf("unmarshal lease query request failed err: %v", err)
+				continue
+			}
+			nodeID, err := messagelayer.GetNodeID(msg)
+			if err != nil {
+				klog.Errorf("failed to get nodeID of lease msg %s, %v", msg.String(), err)
+				continue
+			}
+			resource, err := messagelayer.BuildResource(nodeID, lease.Namespace, model.ResourceTypeLease, lease.Name)
+			if err != nil {
+				klog.Errorf("failed to build resource for lease msg %s, %v", msg.String(), err)
+				continue
+			}
+
+			sendResponseAccordingError := func(err error) {
```

Comment:
for update operation, metamanager will immediately send a response if the msgSource is edged, so the reponse msg here will not be handled by edgecore. 
https://github.com/kubeedge/kubeedge/blob/7b22070881793538aff1234014cf9ed6cb6a6808/edge/pkg/metamanager/process.go#L173-L176